src/NUnitFramework/framework/Internal/PreFilter.cs

```diff
@@ -0,0 +1,208 @@
+// ***********************************************************************
+// Copyright (c) 2014-2018 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace NUnit.Framework.Internal
+{
+    /// <summary>
+    /// The PreFilter class implements a simplified filter for use in deciding which
+    /// Types and Methods should be used to generate tests. It is consructed with a 
+    /// list of strings, each of which may end up being interpreted in various ways.
+    /// </summary>
+    public class PreFilter
+    {
+        private static readonly char[] ARG_START = new char[] { '(', '<' };
+
+        private List<FilterElement> _filters = new List<FilterElement>();
+
+        /// <summary>
+        /// Return a new PreFilter, without elements, which is considered
+        /// empty and always matches.
+        /// </summary>
+        public static PreFilter Empty
+        {
+            get { return new PreFilter(); }
```

Comment:
What do you think about making this into a method like `CreateEmpty()` since you're receiving an instance which is not _inherently_ empty (it can be mutated)?

Reply:
Logically, I agree. However, we use Empty this way in two or three other classes, so I was trying for consistency. See TestFilter.Empty.

Reply:
That's reasonable, even more so if the class is internal. 😇